packages/tasit-sdk/src/ethers.test.js

```diff
@@ -0,0 +1,176 @@
+
+		expect(signedTx).to.equal(expectedSignedTx);
+	});
+
+	it("should sign a message", async function() {
+		const rawMsg = "Hello World!";
+		const signedMsg = await wallet.signMessage(rawMsg);
+
+		const expectedSignedMsg =
+			"0xcdd313d56ac1945e799a8c9d28d971fe40231ab354178611fbfe9b5ec00f493a4e33" +
+			"2781e2e72c1978b0d5540ff9fb667c54dc5c0792d9cf9b0ec10485da0d021b";
+
+		expect(signedMsg).to.be.equals(expectedSignedMsg);
+	});
+
+	it("should sign a binary message", async function() {
+		// The 66 character hex string MUST be converted to a 32-byte array first!
+		const hash =
+			"0x3ea2f1d0abf3fc66cf29eebb70cbd4e7fe762ef8a09bcc06c8edf641230afec0";
+
+		const binData = ethers.utils.arrayify(hash);
+
+		const expectedSignedBinData =
+			"0x4d6e6ed24078bbdb3e2d45a01276666c64f396915050271cc40e5874448b5e876ea6" +
+			"44cb1887e70cc6acd0cf6bfc13146261748c5450d8e21f8980ac7defda811b";
+
+		const signedBinData = await wallet.signMessage(binData);
+
+		expect(signedBinData).to.equal(expectedSignedBinData);
+	});
+
+	const ensSample = {
+		provider: ethers.getDefaultProvider(),
+		name: "registrar.firefly.eth",
+		address: "0x6fC21092DA55B392b045eD78F4732bff3C580e2c",
+	};
+
+	it("should resolve ENS name", async function() {
+		const address = await ensSample.provider.resolveName(ensSample.name);
+		expect(address).to.equal(ensSample.address);
+	});
+
+	it("should lookup ENS address", async function() {
+		const name = await ensSample.provider.lookupAddress(ensSample.address);
+		expect(name).to.equal(ensSample.name);
+	});
+
+	it("should instantiate a contract object", async function() {
+		const contractABI = [
+			"event ValueChanged(address indexed author, string oldValue, string newValue)",
+			"constructor(string value)",
+			"function getValue() view returns (string value)",
+			"function setValue(string value)",
+		];
+		const contractAddress = "0x2bD9aAa2953F988153c8629926D22A6a5F69b14E";
+		contract = new ethers.Contract(contractAddress, contractABI, wallet);
+		expect(contract.address).to.be.equals(contractAddress);
+	});
+});
+
+describe("ethers.js - slow test cases", function() {
+	xit("should get/set contract's value", async function() {
+		const currentValue = await contract.getValue();
+		const message = `I like dogs ${randomWallet.mnemonic}`;
+		expect(currentValue).to.be.not.equals(message);
+
+		const updateValueTx = await contract.setValue(message);
+		await ropstenProvider.waitForTransaction(updateValueTx.hash);
+
+		const newValue = await contract.getValue();
+
+		expect(newValue).to.equal(message);
+	});
+
+	xit("should watch contract's ValueChanged event", async function() {
+		const oldValue = await contract.getValue();
+		const newValue = `I like cats ${randomWallet.mnemonic}`;
+
+		const tx = await contract.setValue(newValue);
+
+		await waitForEvent("ValueChanged", [wallet.address, oldValue, newValue]);
+		await ropstenProvider.waitForTransaction(tx.hash);
+	});
+
+	it("should broadcast signed tx", async function() {
```

Comment:
Does this one need an xit too? If not, why not?

Reply:
Sorry, xits are there by mistake. Do you think that is better they are xits by default?

Reply:
I think until we move to ganache and make them run faster, having timeouts enabled for the test suite and keeping these as xit's is the way to go.

Reply:
Done